src/gt4py/next/iterator/type_system/inference.py

```diff
@@ -564,10 +568,21 @@ def visit_FunCall(
             assert isinstance(tuple_.type, ts.TupleType)
             return tuple_.type.types[index]
 
+        # Collect shift_results and pass them as a kwarg to TypeSynthesizer:applied_as_fieldop.
+        # The node is available here, but not within the TypeSynthesizer, so we handle it here.
+        shifts = {}
+        if is_applied_as_fieldop(node):
+            stencil = node.fun.args[0]  # type: ignore[attr-defined] # Assured by if above
+            referenced_fun_names = symbol_ref_utils.collect_symbol_refs(stencil)
+            shifts["shift_results"] = (
+                trace_shifts.trace_stencil(stencil, num_args=len(node.args))
+                if not referenced_fun_names
+                else []
+            )
```

Comment:
Maybe I'm missing something, but wouldn't it be simpler to directly define a `shift_results` list here:
```suggestion
        shift_results = []
        if is_applied_as_fieldop(node):
            stencil = node.fun.args[0]  # type: ignore[attr-defined] # Assured by if above
            referenced_fun_names = symbol_ref_utils.collect_symbol_refs(stencil)
            if not referenced_fun_names:
                shift_results = trace_shifts.trace_stencil(stencil, num_args=len(node.args))
```

and then use it in the call as:
```python
result = fun(*args, shift_results=shift_results, offset_provider_type=self.offset_provider_type)
```
?